File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        map<char,int>mp,mp1;
        for(auto i:s) mp[i]++;
        for(auto i:mp){
            if(i.second>=2) mp1[i.first]++;
        }
        int k=0;
        for(auto i:s){
            if(!mp1[i]) return k;
            else k++;
        }
        return -1;
    }
    
};

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        dp=defaultdict(int)
        for i in s:
            dp[i]+=1
        k=0
        for i in s:
            if dp[i]==1:
                return k
            k+=1
        return -1